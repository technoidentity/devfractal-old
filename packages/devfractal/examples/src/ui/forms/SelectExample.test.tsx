import React from 'react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'
import { Select } from './SelectExample'

test('Select Example', () => {
  const component: ReactTestRenderer = renderer.create(<Select />)
  expect(component.toJSON()).toMatchSnapshot()
})
