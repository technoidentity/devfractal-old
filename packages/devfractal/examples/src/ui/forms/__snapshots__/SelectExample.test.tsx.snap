// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Example 1`] = `
Array [
  <section
    className="section"
  >
    <h1
      className="title is-4"
    >
      Dynamic Select(
      red
      )
    </h1>
    <div
      className="control"
    >
      <div
        className="select"
      >
        <select
          className=""
          name="red"
          onChange={[Function]}
        >
          <option
            value="red"
          >
            Red
          </option>
          <option
            value="blue"
          >
            Blue
          </option>
          <option
            value="green"
          >
            Green
          </option>
          <option
            value="purple"
          >
            Purple
          </option>
        </select>
      </div>
    </div>
  </section>,
  <section
    className="section"
  >
    <h1
      className="title is-4"
    >
      Select
    </h1>
    <div
      className="control"
    >
      <div
        className="select"
      >
        <select
          className=""
        >
          <option>
            Red
          </option>
          <option>
            Blue
          </option>
          <option>
            Green
          </option>
          <option>
            Purple
          </option>
        </select>
      </div>
    </div>
  </section>,
  <section
    className="section"
  >
    <div
      className="field"
    >
      <h1
        className="title is-4"
      >
        Colors
      </h1>
      <div
        className="control"
      >
        <div
          className="select is-primary"
        >
          <select
            className=""
          >
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="control"
      >
        <div
          className="select is-info"
        >
          <select
            className=""
          >
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="control"
      >
        <div
          className="select is-success"
        >
          <select
            className=""
          >
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="control"
      >
        <div
          className="select is-warning"
        >
          <select
            className=""
          >
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="control"
      >
        <div
          className="select is-danger"
        >
          <select
            className=""
          >
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="section"
  >
    <div
      className="field"
    >
      <h1
        className="title is-4"
      >
        Sizes
      </h1>
      <div
        className="control is-small"
      >
        <div
          className="select is-small"
        >
          <select
            className=""
          >
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="control is-medium"
      >
        <div
          className="select is-medium"
        >
          <select
            className=""
          >
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="control is-large"
      >
        <div
          className="select is-large"
        >
          <select
            className=""
          >
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="control"
      >
        <div
          className="select"
        >
          <select
            className=""
          >
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="section"
  >
    <div
      className="field"
    >
      <h1
        className="title is-4"
      >
        Style
      </h1>
      <div
        className="control"
      >
        <div
          className="select is-rounded"
        >
          <select
            className=""
          >
            <option>
              Rounded dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="section"
  >
    <div
      className="field"
    >
      <h1
        className="title is-4"
      >
        States
      </h1>
      <div
        className="field"
      >
        <div
          className=""
        >
          Normal
        </div>
        <div
          className="control"
        >
          <div
            className="select"
          >
            <select
              className="is-focused"
            >
              <option>
                Select dropdown
              </option>
              <option>
                With options
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        className=""
      >
        Hover
      </div>
      <div
        className="control"
      >
        <div
          className="select"
        >
          <select
            className="is-hovered"
          >
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className=""
      >
        Focus
      </div>
      <div
        className="control"
      >
        <div
          className="select"
        >
          <select
            className="is-focused"
          >
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className=""
      >
        Loading
      </div>
      <div
        className="control"
      >
        <div
          className="select is-loading"
        >
          <select
            className=""
          >
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="section"
  >
    <div
      className="field"
    >
      <h1
        className="title is-4"
      >
        With icons
      </h1>
      <div
        className="control is-small has-icons-left"
      >
        <div
          className="select is-small"
        >
          <select
            className=""
          >
            <option>
              Country
            </option>
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
        <span
          className="icon is-left"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-globe fa-w-16 "
            data-icon="globe"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="control has-icons-left"
      >
        <div
          className="select"
        >
          <select
            className=""
          >
            <option>
              Country
            </option>
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
        <span
          className="icon is-left"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-globe fa-w-16 "
            data-icon="globe"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="control is-medium has-icons-left"
      >
        <div
          className="select is-medium"
        >
          <select
            className=""
          >
            <option>
              Country
            </option>
            <option>
              Select dropdown
            </option>
            <option>
              With options
            </option>
          </select>
        </div>
        <span
          className="icon is-left"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-globe fa-w-16 "
            data-icon="globe"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M336.5 160C322 70.7 287.8 8 248 8s-74 62.7-88.5 152h177zM152 256c0 22.2 1.2 43.5 3.3 64h185.3c2.1-20.5 3.3-41.8 3.3-64s-1.2-43.5-3.3-64H155.3c-2.1 20.5-3.3 41.8-3.3 64zm324.7-96c-28.6-67.9-86.5-120.4-158-141.6 24.4 33.8 41.2 84.7 50 141.6h108zM177.2 18.4C105.8 39.6 47.8 92.1 19.3 160h108c8.7-56.9 25.5-107.8 49.9-141.6zM487.4 192H372.7c2.1 21 3.3 42.5 3.3 64s-1.2 43-3.3 64h114.6c5.5-20.5 8.6-41.8 8.6-64s-3.1-43.5-8.5-64zM120 256c0-21.5 1.2-43 3.3-64H8.6C3.2 212.5 0 233.8 0 256s3.2 43.5 8.6 64h114.6c-2-21-3.2-42.5-3.2-64zm39.5 96c14.5 89.3 48.7 152 88.5 152s74-62.7 88.5-152h-177zm159.3 141.6c71.4-21.2 129.4-73.7 158-141.6h-108c-8.8 56.9-25.6 107.8-50 141.6zM19.3 352c28.6 67.9 86.5 120.4 158 141.6-24.4-33.8-41.2-84.7-50-141.6h-108z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
  </section>,
]
`;
